pkg/login/social/azuread_oauth.go

```diff
@@ -130,6 +134,11 @@ func (s *SocialAzureAD) UserInfo(ctx context.Context, client *http.Client, token
 	}
 	s.log.Debug("AzureAD OAuth: extracted groups", "email", email, "groups", fmt.Sprintf("%v", groups))
 	if !s.isGroupMember(groups) {
+		if len(groups) == 0 {
+			// either they do not have a group or misconfiguration
+			return nil, errAzureADMissingGroups
+		}
+		// a group found in the claims is not in the allowed groups
```

Comment:
I think this comment is wrong, here is my suggestion:
```suggestion
		// no group from the allowed groups was found in the claims
```

Reply:
Or better:
```suggestion
		// user is not a member of any of the allowed groups
```